internal/uvm/vsmb.go

```diff
@@ -32,31 +34,54 @@ func (uvm *UtilityVM) AddVSMB(ctx context.Context, hostPath string, guestRequest
 
 	uvm.m.Lock()
 	defer uvm.m.Unlock()
-	share, err := uvm.findVSMBShare(ctx, hostPath)
+
+	st, err := os.Stat(hostPath)
+	if err != nil {
+		return err
+	}
+	var file string
+	m := uvm.vsmbDirShares
+	if !st.IsDir() {
+		m = uvm.vsmbFileShares
+		file = hostPath
+		hostPath = filepath.Clean(filepath.Dir(hostPath))
```

Comment:
I wonder if we should do the `filepath.Clean` in all cases so that a dir `hostPath` will have the same key?

Reply:
Good idea